src/plugins/flow.js

```diff
@@ -21,6 +21,45 @@ pp.flowParseTypeInitialiser = function (tok, allowLeadingPipeOrAnd) {
   return type;
 };
 
+pp.flowParsePredicate = function() {
+  let node = this.startNode();
+  let moduloLoc = this.state.startLoc;
+  let moduloPos = this.state.start;
+  this.expect(tt.modulo);
+  let checksLoc = this.state.startLoc;
+  this.expectContextual("checks");
+  // Force '%' and 'checks' to be adjacent
+  if (moduloLoc.line !== checksLoc.line || moduloLoc.column !== checksLoc.column - 1) {
+    this.raise(moduloPos, "Unexpected token");
```

Comment:
You could also use `this.unexpected(moduloPos);` here. It's a little bit shorter but otherwise isn't really important if not.